### PR TITLE
Key on observations right hand panel.

### PR DESCRIPTION
**Pull request by rpiaggio**
This seems to solve the issue of the observation editor not changing with the selected observation.

### CODE REVIEW

File: explore/src/main/scala/explore/config/ConfigurationPanel.scala [hunk 1]
```diff
@@ -152,12 +152,14 @@ object ConfigurationPanel {
   // TODO Move these to react-datetime
   implicit class InstantOps(val instant: Instant) extends AnyVal {
     // DatePicker only works in local timezone, so we trick it by adding the timezone offset.
+    // See https://github.com/Hacker0x01/react-datepicker/issues/1787
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/config/ConfigurationPanel.scala:155]: Unrelated, just added more info to comment.

File: explore/src/main/scala/explore/tabs/AsterismEditorTile.scala [hunk 2]
```diff
@@ -47,11 +47,12 @@ object AsterismEditorTile {
     width:           Int,
     height:          Int
   )(implicit ctx:    AppContextIO) =
-    Tile(ObsTabTiles.TargetId,
-         title,
-         back = backButton,
-         canMinimize = true,
-         bodyClass = Some(ExploreStyles.TargetTileBody)
+    Tile(
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/AsterismEditorTile.scala:50]: Unrelated, just formatted.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 3]
```diff
@@ -407,7 +407,7 @@ object ObsTabContents {
             clazz = ExploreStyles.ObservationTiles.some
           ): VdomNode
         }
-      )
+      ).withKey(obsId.toString)
```
Comment (toddburnside) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:410]: Wow, that did it, eh? Nice.

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**